src/main/java/org/opentripplanner/middleware/utils/ItineraryUtils.java

```diff
@@ -176,6 +178,42 @@ public static boolean itinerariesMatch(Itinerary referenceItinerary, Itinerary c
         return true;
     }
 
+    /**
+     * Checks that the specified itinerary is on the same day as the specified date/time.
+     * @param itinerary the itinerary to check.
+     * @param requestDateTime the request date/time to check, in the OTP's time zone.
+     * @param tripIsArriveBy true to check the itinerary endtime, false to check the startTime.
+     * @return true if the itinerary's startTime or endTime is one the same day as the day of the specified date and time.
+     */
+    public static boolean isSameDay(Itinerary itinerary, ZonedDateTime requestDateTime, boolean tripIsArriveBy) {
+        // TODO: Make SERVICE_DAY_START_HOUR an optional config parameter.
+        final int SERVICE_DAY_START_HOUR = 3;
```

Comment:
I'd go ahead and make this an optional config param.

Reply:
It is an optonal config param in https://github.com/ibi-group/otp-middleware/pull/102/commits/911c618356112c4db4b089e158aab92ab6c23062.